prog/vnet/load_balancer_health_probes.rb

```diff
@@ -10,8 +10,13 @@ def vm
   label def health_probe
     response_code = nil
     begin
-      endpoint = "#{vm.nics.first.private_ipv4.network}:#{load_balancer.dst_port}#{load_balancer.health_check_endpoint}"
-      response_code = vm.vm_host.sshable.cmd("sudo ip netns exec #{vm.inhost_name} curl --max-time #{load_balancer.health_check_timeout} --silent --output /dev/null --write-out '%{http_code}' #{endpoint}")
+      cmd = if load_balancer.health_check_protocol == "tcp"
+        "sudo ip netns exec #{vm.inhost_name} nc -z -w #{load_balancer.health_check_timeout} #{vm.nics.first.private_ipv4.network} #{load_balancer.dst_port} && echo 200 || echo 400"
+      else
+        "sudo ip netns exec #{vm.inhost_name} curl --insecure --max-time #{load_balancer.health_check_timeout} --silent --output /dev/null --write-out '%{http_code}' #{load_balancer.health_check_protocol}://#{vm.nics.first.private_ipv4.network}:#{load_balancer.dst_port}#{load_balancer.health_check_endpoint}"
```

Comment:
Should we use `--insecure` for `https` protocol?

Reply:
I think for a health check it's okay, to allow for self-signed certificates or expired certificates. A "nice" feature would be to tell the customer about expiry :)

I am not a huge fan of using curl altogether though. Memory-unsafe C program that has fork overhead when we don't really need either demerit. But I think @furkansahin was thinking he could complicate things and make them more efficient via SSH tunnels in the monitor process later.

Reply:
`--insecure` is OK because we're already inside network namespace. Even if the customer did not get the certificate, we want the load balancer to perform health checks and decide that this is healthy. This is the only way they can realise the connection errors are about the certificate but not the end to end connection they have from client to the server.